flink-end-to-end-tests/test-scripts/common_docker.sh

```diff
@@ -33,8 +33,25 @@ function containers_health_check() {
   done
 }
 
-function build_image_with_jar() {
-    local job_artifacts=$1
-    local image_name=${2:-flink-job}
-    ./build.sh --from-local-dist --job-artifacts ${job_artifacts} --image-name ${image_name}
+function build_image() {
+    local image_name=${1:-flink-job}
+
+    echo "Starting fileserver for Flink distribution"
```

Comment:
maybe, I'm not so sure about it. On the one hand there are some environmental dependencies (see above) and you would need to harden it way more to ensure cleanup of resources (which I'd prefer not to do atm).

Reply:
In my suggestion above, we only require docker
For the cleanup, we could build the archive in `/tmp`.

Reply:
My concern is not the cleanup of files; the archive and git clone are placed under `TEST_DATA_DIR` which is cleaned up anyway.

I'm only concerned about the fileserver process; you definitely don't want that one leaking, since it could affect the correctness of subsequent runs.

So as I said, it would need more work to be in a state where I would offer it to other people.